Dynatrace/dynatrace-operator
# Reworks imageInstaller to extract into a shared dir

# Description

The current way we download/cache and unpack  `codeModulesImage` is not perfect when it comes to node storage usage, as it extracts the image for every tenant.

This PR changes how the provisioner prepares the node for the driver.
The image is downloaded to `/cache/<image-ingest>` and gets unpacked to `data/codemodules/<image-digest>`.
We only download/unpack if the digest changes (or its a new digest), we store the digest in the `dynakubes` table in the metadata database on the node.
After we unpack to `data/codemodules/<image-digest>` then `/cache/<image-ingest>` gets cleaned up as its no longer needed for anything.

Because of the shared nature of the agent "binary" we can't update the `ruxitagentproc.conf` at 
`data/codemodules/<image-digest>`, so we instead create a `data/<tenant>/config/...` where we update/merge the necessary config files.

The driver WILL use `data/codemodules/<image-digest>` + `data/<tenant>/config/...` as 2 of its lowerdirs during mounting the overlayfs volume for user pods.
Therefore `data/<tenant>/config/...` needs to follow the same folder structure as `data/codemodules/<image-digest>`.

## How can this be tested?
Deploy anything that uses the csidriver + `codeModulesImage`, 
then check make sure the csidriver works by: 
- seeing no error in the logs for the provisioner (the driver will not work which is expected)
- check the filesystem for:
  - `data/codemodules/<image-digest>` is a valid oneagent "install"
  - `data/codemodules/<image-digest>/agent/bin/current` is a valid symlink
  - `data/<tenant>/config/agent/conf/ruxitagentproc.conf` is a valid "tenantified" conf file
  - `/cache/<image-ingest>`is cleaned up 



## Checklist
